src/compiler/transformers/module/module.ts

```diff
@@ -1613,16 +1625,27 @@ export function transformModule(context: TransformationContext): (x: SourceFile
             for (const specifier of node.exportClause.elements) {
                 const exportNeedsImportDefault = !!getESModuleInterop(compilerOptions) &&
                     !(getInternalEmitFlags(node) & InternalEmitFlags.NeverApplyImportHelper) &&
-                    idText(specifier.propertyName || specifier.name) === "default";
-                const exportedValue = factory.createPropertyAccessExpression(
-                    exportNeedsImportDefault ? emitHelpers().createImportDefaultHelper(generatedName) : generatedName,
-                    specifier.propertyName || specifier.name,
-                );
+                    moduleExportNameIsDefault(specifier.propertyName || specifier.name);
+                const specifierName = specifier.propertyName || specifier.name;

```

Comment:
I would reorder these lines so that `const specifierName` comes first. Then you can reuse `specifierName` in the call to `moduleExportNameIsDefault`.